keadm/app/cmd/cloud/init.go

```diff
@@ -101,8 +114,24 @@ func addJoinOtherFlags(cmd *cobra.Command, initOpts *types.InitOptions) {
 		"Use this key to download and use the required Kubernetes version")
 	cmd.Flags().Lookup(types.KubernetesVersion).NoOptDefVal = initOpts.KubernetesVersion
 
-	cmd.Flags().StringVar(&initOpts.KubeConfig, types.KubeConfig, initOpts.KubeConfig,
-		"Use this key to set kube-config path, eg: $HOME/.kube/config")
+	cmd.Flags().StringVar(&initOpts.ControllerYaml.Controller.Kube.Master, types.KubeMaster, initOpts.ControllerYaml.Controller.Kube.Master,
+		"Use this key to set controller.kube.master , eg: http://localhost:8080")
+
+	cmd.Flags().StringVar(&initOpts.ControllerYaml.Controller.Kube.KubeConfig, types.ControllerKubeKubeconfig, initOpts.ControllerYaml.Controller.Kube.KubeConfig,
+		"Use this key to set controller.kube.kubeconfig path, eg: $HOME/.kube/config")
+
+	cmd.Flags().StringVar(&initOpts.ControllerYaml.DeviceController.Kube.Master, types.DeviceControllerKubeConfig, initOpts.ControllerYaml.DeviceController.Kube.KubeConfig,
```

Comment:
I think we can optimize by making kube config as a common configuration in the yaml .  I am not sure if we have an open issue for it.

Reply:
@rohitsardesai83  i agree with you.    

@sids-b @samy2019  what do you think?